UACloudLibraryServer/CloudLibDataProvider.cs

```diff
@@ -3,56 +3,110 @@
 using System.Globalization;
 using System.Linq;
 using System.Linq.Expressions;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 using CESMII.OpcUa.NodeSetModel;
 using Extensions;
 using HotChocolate.Data;
 using HotChocolate.Types;
+using HotChocolate;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
 using Opc.Ua.Cloud.Library.DbContextModels;
+using Opc.Ua.Cloud.Library.Models;
 
 namespace Opc.Ua.Cloud.Library
 {
-    public class CloudLibDataProvider
+    public class CloudLibDataProvider : IDatabase
     {
-        private readonly AppDbContext _context = null;
+        private readonly AppDbContext _dbContext = null;
         private readonly UaCloudLibResolver _resolver = null;
+        private readonly ILogger _logger;
 
-        public CloudLibDataProvider(AppDbContext context, UaCloudLibResolver resolver)
+        public CloudLibDataProvider(AppDbContext context, UaCloudLibResolver resolver, ILoggerFactory logger)
         {
-            _context = context;
+            _dbContext = context;
             _resolver = resolver;
```

Comment:
Can we also get rid of UaCloudLibResolver?

Reply:
Ok, will merge it into the CloudLibDataProvider.cs. We can't get rid of some of the functionality in there (ApplyWhere) until we remove the old (marked as obsolete) GraphQL collections.

Reply:
Done